plugins/vpc-bridge/config/k8s/connector.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+//go:build enablek8sconnector && windows
+
+package k8s
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+
+	log "github.com/cihub/seelog"
+)
+
+const (
+	// Represents Env variable for k8s connector binary path.
+	envK8sConnectorBinaryPath = "AWS_VPC_CNI_K8S_CONNECTOR_BINARY_PATH"
+	// Represents default path for k8s connector binary in EKS Windows AMIs.
+	defaultK8sConnectorBinaryPath = `C:\Program Files\Amazon\EKS\bin\aws-vpc-cni-k8s-connector.exe`
+
+	// Represents Env for log level for k8s connector binary.
+	envLogLevel = "VPC_CNI_LOG_LEVEL"
+	// Represents default log level for k8s connector binary.
+	defaultLogLevel = "info"
+)
+
+// GetPodIP retrieves pod IP using k8s connector binary.
+// Output from binary is received over named pipe.
+// Create and read operations on named pipe are handled in separate go routine.
+// Named pipe path will be passed as argument to k8s connector binary execution.
+func GetPodIP(ctx context.Context, podNamespace, podName string) (string, error) {
+
+	pipeName, err := newPipe()
+	if err != nil {
+		return "", fmt.Errorf("error creating new pipe: %w", err)
+	}
+
+	resultChan := make(chan pipeReadResult)
+
+	// Read output from named pipe in separate go routine as accepting and reading connection is blocking operation.
+	go readResultFromPipe(ctx, pipeName, resultChan)
+
+	// Executing k8s connector binary in main routine. binary will write output to named pipe.
+	err = executeK8sConnector(ctx, podNamespace, podName, pipeName)
+	if err != nil {
+		return "", fmt.Errorf("error executing k8s connector: %w", err)
+	}
+
+	// Get output from named pipe using result chan.
+	var result pipeReadResult
+	select {
+	// Check if context timed out.
+	case <-ctx.Done():
+		return "", fmt.Errorf("error getting output from pipe: %w", ctx.Err())
+	case result = <-resultChan:
+	}
+
+	if result.error != nil {
+		return "", fmt.Errorf("error reading output from pipe: %w", result.error)
+	}
+
+	log.Debugf("Got pod IP %s for pod %s in namespace %s", result.output, podName, podNamespace)
```

Comment:
"IP" -> "IP address". Multiple occurrences in this PR.

Reply:
Addressed